docs/configuration/http_filters/grpc_json_transcoder_filter.rst

```diff
@@ -0,0 +1,42 @@
+.. _config_http_filters_grpc_json_transcoding:
+
+gRPC-JSON transcoder filter
+===========================
+
+gRPC :ref:`architecture overview <arch_overview_grpc>`.
+
+This is a filter which allows a RESTful JSON API client to send requests to Envoy over HTTP
+and get proxied to a gRPC service. The HTTP mapping for the gRPC service has to be defined by
+`custom options <https://cloud.google.com/service-management/reference/rpc/google.api#http>`_.
+
+Configure gRPC-JSON transcoder
+------------------------------
+
+To configure gRPC-JSON transcoder, you'll need to generate protobuf descriptors from your gRPC
+service. You'll also need to clone the googleapis repository from Github before running protoc,
+as you'll need annotations.proto in your include path.
+
+.. code-block:: bash
+
+  git clone https://github.com/googleapis/googleapis
+  GOOGLEAPIS_DIR=<your-local-googleapis-folder>
+
+Then run protoc to generate .pb descriptor from bookstore.proto:
+
+.. code-block:: bash
+
+  protoc -I$(GOOGLEAPIS_DIR) -I. --include_imports --include_source_info \
+    test/proto/bookstore.proto --descriptor_set_out=proto.pb
+
+The filter config for the filter need the descriptor file, and the services you want to enable.
+
+.. code-block:: json
+
+  {
+    "type": "both",
+    "name": "grpc_json_transcoder",
+    "config": {
+      "proto_descriptors": "proto.pb",
```

Comment:
> Can we get a description of each of these config parameters? If I understand correctly, it's required that all descriptors get merged into a single pb file, right? What if there are multiple services from multiple protos? How do those get merged? Can you maybe provide an example?

Will write it in docs, in short, protoc accepts multiple input files, so just do them together.

> For "services" what happens if the .pb file contains more services than you specify? Those aren't transcoded? Can you clarify?

Right, those aren't transcoded.

> Also, in general, I would love to talk about reflection possibilities. Obviously this is a follow up, but from Lyft perspective (and I think others), this filter is vastly less useful and more difficult to use if Envoy has to know ahead of time the .pb files and we need to keep them up to date. If we could do this with reflection I think that is the true absolutely killer feature.

I totally agree with you that would provide better usability, I'm just saying that won't included in this PR as this is big enough :) Feel free to open an issue and assign to me once this get merged.